src/coreclr/vm/comsynchronizable.cpp

```diff
@@ -1149,6 +1149,40 @@ extern "C" BOOL QCALLTYPE ThreadNative_YieldThread()
     return ret;
 }
 
+extern "C" BOOL QCALLTYPE ThreadNative_Abort(QCall::ThreadHandle thread, int timeout)
+{
+    QCALL_CONTRACT;
+
+    BOOL ret = FALSE;
+
+    BEGIN_QCALL
+
+    ret = thread->UserAbort(EEPolicy::TA_Safe, (DWORD)timeout) == S_OK;
+
+    END_QCALL
+
+    return ret;
+}
+
+// Unmarks the current thread for abort.
+// Returns true if the thread had the TS_AbortRequested flag set.
+FCIMPL0(FC_BOOL_RET, ThreadNative::ResetAbort)
```

Comment:
Would it be worthwhile to make this a "suppress GC Transition QCall" instead of an FCall?

Reply:
Does that also suppress checking the `TS_AbortRequested` flag before performing a `QCALL`?  Normal `QCALL`s are aborted by this flag.

Reply:
I need to check that. I don’t think we erect a HelperMethodFrame (and I know we don’t erect an InlinedCallFrame), so I think the check is suppressed, but I’m not 100% sure.

Reply:
We don't erect any frames, so you should be able to do this as a "suppress GC Transition QCall" instead of an FCall. Normal QCalls are aborted because we check the flag when returning from a P/Invoke where we transitioned. If we don't transition, we never set up the frame to check.

Reply:
I can try that.  What are benefits of making this a QCall?

Reply:
Mainly less overhead than an FCall (there's no work in the macros. It's just a simple call with no transition cost).

Additionally, we have a long-term goal to move many if not all FCalls to be either QCalls or suppress-GC-transition QCalls as they're in general less error prone. The only ones that will stay as FCalls are ones that absolutely need to for performance (if none exist, we can remove the concept of an FCall from CoreCLR).